### PR TITLE
Redirect testing.md to how-to-test-workflow-definitions-in-go.md

### PR DESCRIPTION
**Pull request by aarohib**
## What does this PR do?

Redirects testing.md to a new how-to-test-workflow-definitions-in-go.md, with added detail about SetTestTimeout per #489

### CODE REVIEW

File: static/_redirects [hunk 1]
```diff
@@ -57,8 +57,8 @@
 /docs/go-sync-vs-async-start        /docs/go/sync-vs-async-start
 /docs/go/sync-vs-async-start        /docs/go/workflows/#how-to-start-a-workflow
 /docs/go-task-queues        /docs/go/task-queues
-/docs/go-testing        /docs/go/testing
-/docs/go-workflow-testing         /docs/go/testing
+/docs/go-testing        /docs/go/how-to-test-workflow-definitions-in-go
```
Comment (flossypurse) [inline static/_redirects:60]:
```suggestion
/docs/go/testing        /docs/go/how-to-test-workflow-definitions-in-go
```

`/docs/go-testing` is so old now, I think we can just focus on redirecting `/docs/go/testing`

File: static/_redirects [hunk 2]
```diff
@@ -57,8 +57,8 @@
...
+/docs/go-workflow-testing         /docs/go/how-to-test-workflow-definitions-in-go
```
Comment (flossypurse) [inline static/_redirects:61]:
```suggestion
```
We can even delete this one entirely at this point.

File: static/_redirects (hunk 1 above)
Comment (aarohib) [reply static/_redirects]: Updated

File: static/_redirects (hunk 2 above)
Comment (aarohib) [reply static/_redirects]: Updated

File: docs/go/how-to-test-workflow-definitions-in-go.md [hunk 3]
```diff
@@ -92,6 +92,8 @@ Next, we implement a `SetupTest` method to set up a new test environment before
 ensures that each test runs in its own isolated sandbox. We also implement an `AfterTest` function
 where we assert that all mocks we set up were indeed called by invoking `s.env.AssertExpectations(s.T())`.
 
+Timeout for the entire test can be set using `SetTestTimeout` in the Workflow or Activity environment.
```
Comment (flossypurse) [inline docs/go/how-to-test-workflow-definitions-in-go.md:95]: Any chance there is an example of how to set that?

**Review by flossypurse — APPROVED** [review]

**Pull request merged**